handlers/deploy.go

```diff
@@ -150,23 +150,16 @@ func deploy(ctx context.Context, req types.FunctionDeployment, client *container
 		return errors.Wrapf(err, "failed to setup network for namespace %q: %v", id, err)
 	}
 
-	// Get the IP of the default interface.
-	defaultInterface := gocni.DefaultPrefix + "0"
-
-	if _, ok := result.Interfaces[defaultInterface]; !ok {
-		return fmt.Errorf("failed to find interface %q", defaultInterface)
-	}
-	if result.Interfaces[defaultInterface].IPConfigs != nil &&
-		len(result.Interfaces[defaultInterface].IPConfigs) == 0 {
-		return fmt.Errorf("failed to find IP for interface %q, no configs found", defaultInterface)
+	// Get the IP of the created interface
+	for _, config := range result.Interfaces {
+		if config.Sandbox == netns {
+			for _, ipConfig := range config.IPConfigs {
+				serviceMap.Add(name, &ipConfig.IP)
+				log.Printf("%s has IP: %s.\n", name, &ipConfig.IP)
+			}
+		}
```

Comment:
Let's discover the IP in the loop, then act on it after the loop, either from breaking out or from finishing without a result.